src/main/java/de/teamlapen/vampirism/entity/action/EntityActionHandler.java

```diff
@@ -0,0 +1,291 @@
+                actionsMap.compute(EntityActions.entity_heal, (k, v) -> v + 2);
+            else if (healthPercent < 0.4)
+                actionsMap.compute(EntityActions.entity_heal, (k, v) -> v + 1);
+        } else {
+            /* Regeneration */
+            if (healthPercent < 0.1)
+                actionsMap.computeIfPresent(EntityActions.entity_regeneration_areaofeffect, (k, v) -> v + 2);
+            else if (healthPercent < 0.4)
+                actionsMap.computeIfPresent(EntityActions.entity_regeneration_areaofeffect, (k, v) -> v + 1);
+        }
+
+        if (entity instanceof IVampire) {
+            /* Vampire Only Actions */
+            IVampire entity = (IVampire) this.entity;
+            /* Sunscream Action */
+            if (entity.isGettingSundamage() && !entity.isIgnoringSundamage()) {
+                actionsMap.computeIfPresent(EntityActions.entity_sunscreen, (k, v) -> v + actionsMap.size() < 5 ? 4 : 2);
+            }
+            /* Dark Projectile Action */
+            if (distanceToTarget > 20)
+                actionsMap.computeIfPresent(EntityActions.entity_dark_projectile, (k, v) -> v + 2);
+            else if (distanceToTarget > 10)
+                actionsMap.computeIfPresent(EntityActions.entity_dark_projectile, (k, v) -> v + 1);
+            /* Invisible Action */
+            if (distanceToTarget > 4 && actionsMap.size() < 5)
+                actionsMap.computeIfPresent(EntityActions.entity_invisible, (k, v) -> v + actionsMap.size() < 5 ? 2 : 1);
+        } else if (entity instanceof IHunter) {
+            /* Hunter Only Actions */
+            IHunter entity = (IHunter) this.entity;
+        }
+
+        List<IEntityAction> actionList = new ArrayList<>();
+        for (Map.Entry<IEntityAction, Integer> e : actionsMap.entrySet()) {
+            for (int i = 0; i < e.getValue(); i++) {
+                actionList.add(e.getKey());
+            }
+        }
+        return actionList.get(entity.getRNG().nextInt(actionList.size()));
+    }
+
+    public void handle() {
+        if (availableActions != null && !availableActions.isEmpty()) {
+            if (entity.getAttackTarget() instanceof EntityPlayer) {
+                if (isPlayerTarget) {
+                    updateHandler();
+                } else {
+                    startExecuting();
+                    isPlayerTarget = true;
+                }
+            } else {
+                if (isPlayerTarget) {
+                    isPlayerTarget = false;
+                    deactivateAction();
+                }
+            }
+        }
+    }
+
+    public IEntityAction getAction() {
+        return action;
+    }
+
+    public boolean isPlayerTarget() {
+        return isPlayerTarget;
+    }
+
+    public void readFromNBT(NBTTagCompound nbt) {
+        if (nbt.hasKey("activeAction")) {
+            deactivateAction(VampirismAPI.entityActionManager().getRegistry().getValue(new ResourceLocation("vampirism", nbt.getString("activeAction"))));
```

Comment:
"Vampirism" probably should not be hardcoded here, since in theory other mods could register their own actions.
The modid should be part of the saved nbt string